pkg/apis/core/validation/validation.go

```diff
@@ -2689,6 +2711,20 @@ func validateContainers(containers []core.Container, isInitContainers bool, volu
 	return allErrs
 }
 
+//check that we have at least one container in the pod which is not set to lifecyle type sidecar
+func validateSidecars(containers []core.Container, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+
+	for _, container := range containers {
+		if container.Lifecycle == nil || container.Lifecycle.Type == nil || *container.Lifecycle.Type != core.LifecycleTypeSidecar {
```

Comment:
Why don't we check for `container.Lifecycle.Type == core.LifecycleTypeStandard` instead?

Reply:
yeah seems safer to do that, changed